util/files.py

```diff
@@ -18,28 +22,76 @@ def random_ascii(length, rng=None):
         rng = random
     return ''.join([rng.choice(string.ascii_letters) for _ in range(length)])
 
-def create_submission_dir(course, exercise):
-    '''
-    Creates a directory for a submission.
-
-    @type course: C{dict}
-    @param course: a course configuration
-    @type exercise: C{dict}
-    @param exercise: an exercise configuration
-    @rtype: C{str}
-    @return: directory path
-    '''
-
-    # Create a unique directory name for the submission.
-    d = datetime.datetime.now().strftime("%Y%m%d%H%M%S%f") + random_ascii(5)
-    submission_dir = os.path.join(settings.SUBMISSION_PATH,
-        course["key"], exercise["key"], d)
-
-    # Create empty directory.
-    if not os.path.exists(submission_dir):
-        os.makedirs(submission_dir)
-
-    return submission_dir
+class SubmissionDir:
```

Comment:
The SubmissionDir class is effectively just a copy of the original code but wrapped inside a class. I don't think it is a good idea to spend time on those now. My additions, on the other hand, either catch exceptions or deliberately let them bubble up, so that the calling code knows what went wrong.